scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -130,7 +131,14 @@ private[spel] class Typer(
       current: TypingContext
   ): NodeTypingResult = {
 
-    def toNodeResult(typ: TypingResult) = current.toResult(TypedNode(node, TypingResultWithContext(typ)))
+    // We are converting arrays to lists everywhere
```

Comment:
1. It is not exactly everyhwere - this  `toNodeResult` isn't used in every place. For example, in `typeChildrenNodes`, `TypedNode.apply` is used instead. 
2. This conversion will be used in some certain situations:
a) When a user uses projection/selection - am I right?
b) When a target type (e.g.) of method parameter is a List and we want to put Array into it
c) When user uses method that is not available in a source type
d) ... ?

In other places this type won't be change e.g.
aa) When we just assign array to local variables e.g. `array = "1,2".split(",")`
bb) When we pass array to parameter that is of `Array` type

In these mentioned places we can't change the type because it would cause `ClassCastException` during serialization of object by Flink. I think that we have to extract method in these places and do this conversion only when we are sure that it will be done in runtime as well

Reply:
I agree on this. According to this I changed implementation to handle selections, projects and passing array as parameter of list type.